include/internal/provider.h

```diff
@@ -54,7 +55,7 @@ int ossl_provider_forall_loaded(OPENSSL_CTX *,
 /* Getters for other library functions */
 const char *ossl_provider_name(OSSL_PROVIDER *prov);
 const DSO *ossl_provider_dso(OSSL_PROVIDER *prov);
-const char *ossl_provider_module_name(OSSL_PROVIDER *prov);
+const char *ossl_provider_module_name(const OSSL_PROVIDER *prov);
 const char *ossl_provider_module_path(OSSL_PROVIDER *prov);
```

Comment:
It's a good point to constify this one, but that should probably happen with all the getters

Reply:
yep i will revert this change - it was more of a mental note that it needed to be done.